src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,482 @@
+
+```math
+d_{\mathcal P(n)}(x,y) = \lVert \operatorname{Log}(x^{-\frac{1}{2}}yx^{-\frac{1}{2}})\rVert_{\mathrm{F}}.,
+```
+where $\operatorname{Log}$ denotes the matrix logarithm and $\lVert\cdot\rVert_{\mathrm{F}}$ denotes the
+matrix Frobenius norm.
+"""
+function distance(M::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,y) where N
+    s = real.( eigvals( x,y ) )
+    return any(s .<= eps() ) ? 0 : sqrt(  sum( abs.(log.(s)).^2 )  )
+end
+
+@doc doc"""
+    distance(M,x,y)
+
+computes the distance on the manifold of [`SymmetricPositiveDefinite`](@ref)
+nmatrices, i.e. between two symmetric positive definite matrices `x` and `y`
+with respect to the [`LogCholeskyMetric`](@ref). The formula reads
+
+````math
+d_{\mathcal P(n)}(x,y) = \sqrt{
+ \lVert \lfloor l \rfloor - \lfloor k \rfloor \rVert_{\mathrm{F}}^2
+ + \lVert \log(\operatorname{diag}(l)) - \log(\operatorname{diag}(k))\rVert_{\mathrm{F}}^2 },
+````
+where $l$ and $k$ are the cholesky factors of $x$ and $y$, respectively,
+$\lfloor\cdit\rfloor$ denbotes the lower triangulr matrix of its argument,
+and $\lVert\cdot\rVert_{\mathrm{F}}$ denotes the Frobenius norm.
+"""
+distance(M::MetricManifold{SymmetricPositiveDefinite{N},LogCholeskyMetric},x,y) where N = distance(CholeskySpace{N}(), cholesky(x).L, cholesky(y).L)
+
+@doc doc"""
+    distance(M,x,y)
+
+computes the distance on the [`SymmetricPositiveDefinite`](@ref) manifold between
+`x` and `y` as a [`MetricManifold`](@ref) with [`LogEuclideanMetric`](@ref).
+The formula reads
+
+```math
+    d_{\mathcal P(n)}(x,y) = \lVert \Log x - \Log y \rVert_{\mathrm{F}}
+```
+where $\operatorname{Log}$ denotes the matrix logarithm and $\lVert\cdot\rVert_{\mathrm{F}}$ denotes the
+matrix Frobenius norm.
+"""
+function distance(M::MetricManifold{SymmetricPositiveDefinite{N},LogEuclideanMetric},x,y) where N
+    eX = eigen(Symmetric(x))
+    UX = eX.vectors
+    SX = eX.values
+    eY = eigen(Symmetric(y))
+    UY = eY.vectors
+    SY = eY.values
+    return norm( UX*Diagonal(log.(SX))*transpose(UX) - UY*Diagonal(log.(SY))*transpose(UY))
+end
+
+@doc doc"""
+    inner(M,x,v,w)
+
+compute the inner product of `v`, `w` in the tangent space of `x` on the [`SymmetricPositiveDefinite`](@ref)
+manifold `M`, which defaults to the [`LinearAffineMetric`](@ref).
+"""
+inner(M::SymmetricPositiveDefinite{N}, x, w, v) where N = inner(MetricManifold(M,LinearAffineMetric()),x,w,v)
+
+@doc doc"""
+    inner(M,x,v,w)
+
+compute the inner product of `v`, `w` in the tangent space of `x` on
+the [`SymmetricPositiveDefinite`](@ref) manifold `M`, as
+a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref). The formula reads
+
+```math
+( v, w)_x = \operatorname{tr}(x^{-1}\xi x^{-1}\nu ),
+```
+"""
+function inner(M::MetricManifold{SymmetricPositiveDefinite{N}, LinearAffineMetric}, x, w, v) where N
+    F = factorize(x)
+    return tr( ( Symmetric(w) / F ) * ( Symmetric(v) / F ) )
```

Comment:
I do not know what you mean by `VERSION` branching (sorry), but this was indeed old code so I am fine updating it to your version.

Reply:
Hm, just shortly trying replacing the two lines I get
````julia
  MethodError: no method matching /(::Symmetric{Float32,Array{Float32,2}}, ::Cholesky{Float32,Array{Float32,2}})
````
I think you meant
````julia
F = cholesky(Symmetric(x)).L
````
?

Reply:
I guess you could remove the `Symmetric` wrappers around `v` and `w`. `factorize` wouldn't give only one factor of a factorization, so I really meant `cholesky`.
By the branching, I meant something along the lines
```julia
if VERSION >= v"1.3.0-rc5.1"
    # use the new right-solve for Cholesky objects
else
    # use this code
end
```
It may still be that you're running into errors because there is no method for `/(::Symmetric,::Cholesky)`, and your `w` and `v` may already be `Symmetric`, even without wrapping them. But then I'd again suggest to add this method to `LinearAlgebra`.

Reply:
Ah, right – I am still running Julia 1.2 – so there might not yet be a `/` for my Cholesky+Symmetric?

For the Code-version-branching – for me that feels like it would introduce quite some clutter if one does that regularly, but thanks for clarifying.

Reply:
I agree about the clutter, but that's how it is with a long-term support version, to which by definition new functionality is not added.

Reply:
Ok, I will add your lines when I have the chance to test that with 1.3 (since in 1.2 I don't have `/` for cholesky).